modules/shubin_m_numeric_systems/src/CustomNSNumber.cpp

```diff
@@ -0,0 +1,399 @@
+                    move_digit;
+      res.digits.push_back(temp_number % digit_t_num_sys);
+      move_digit = temp_number / digit_t_num_sys;
+    }
+    if (move_digit > 0) {
+      res.digits.push_back(move_digit);
+    }
+  } else {
+    res.negative = (temp > *this) ? (true) : (false);
+    digit_t temp_number = 0;
+    digit_t this_number = 0;
+    digit_t other_number = 0;
+    digit_t move_digit = 0;
+    for (size_t i = 0; i < std::max(digits.size(), temp.digits.size()); i++) {
+      this_number = (i < digits.size()) ? (digits[i]) : (0);
+      other_number = (i < temp.digits.size()) ? (temp.digits[i]) : (0);
+      temp_number = digit_t_num_sys + ((res.negative == negative) ?
+                    (this_number - other_number) :
+                    (other_number - this_number)) - move_digit;
+      res.digits.push_back(temp_number % digit_t_num_sys);
+      move_digit = (temp_number / digit_t_num_sys >= 1) ? (0) : (1);
+    }
+  }
+
+  digit_t temp_number = 0;
+  digit_t move_digit = 0;
+  for (size_t i = 0; i < res.digits.size(); i++) {
+    temp_number = res.digits[i] + move_digit;
+    res.digits[i] = temp_number % digit_t_num_sys;
+    move_digit = temp_number / digit_t_num_sys;
+  }
+  if (move_digit > 0) {
+    res.digits.push_back(move_digit);
+  }
+
+  if (res.digits.size() == 0) {
+    res.digits.push_back(0);
+  }
+
+  return res;
+}
+
+CustomNSNumber CustomNSNumber::operator*(const CustomNSNumber& _num) {
+  if (num_sys != _num.num_sys) {
+    throw std::invalid_argument("Numeric systems don't match");
+  }
+
+  CustomNSNumber res = CustomNSNumber(num_sys);
+  CustomNSNumber temp = _num;
+  digit_t digit_t_num_sys = static_cast<digit_t>(res.num_sys);
+
+  if (negative != _num.negative) {
+    res.negative = true;
+  } else {
+    res.negative = false;
+  }
+
+  digit_t temp_number = 0;
+  digit_t move_digit = 0;
+  for (size_t i = 0; i < temp.digits.size(); i++) {
+    for (size_t j = 0; j < digits.size(); j++) {
+      temp_number = temp.digits[i] * digits[j] + move_digit;
+      if (i + j < res.digits.size()) {
+        res.digits[i + j] += temp_number % digit_t_num_sys;
+      } else {
+        res.digits.push_back(temp_number % digit_t_num_sys);
+      }
+      move_digit = temp_number / digit_t_num_sys;
+    }
+    if (move_digit > 0) {
+      res.digits.push_back(move_digit);
+      move_digit = 0;
+    }
+  }
+
+  for (size_t i = 0; i < res.digits.size(); i++) {
+    temp_number = res.digits[i] + move_digit;
+    res.digits[i] = temp_number % digit_t_num_sys;
+    move_digit = temp_number / digit_t_num_sys;
+  }
+  if (move_digit > 0) {
+    res.digits.push_back(move_digit);
+  }
+
+  if (res.digits.size() == 0) {
+    res.digits.push_back(0);
+  }
+
+  return res;
+}
+
+bool CustomNSNumber::operator==(const CustomNSNumber& _num) {
+  CustomNSNumber temp = _num;
+  number_t lhs = ToDec();
+  number_t rhs = temp.ToDec();
+  return (lhs == rhs);
+}
```

Comment:
Зачем создавать временный объект?
Лучше пометить метод `ToDec` как `const` и вызвать напрямую при создании `rhs`.
```suggestion
bool CustomNSNumber::operator==(const CustomNSNumber& _num) const {
  number_t lhs = ToDec();
  number_t rhs = num.ToDec();
  return (lhs == rhs);
}
```
а еще лучше так:
```suggestion
bool CustomNSNumber::operator==(const CustomNSNumber& _num) const {
  return ToDec() == num.ToDec();
}
```
Также поправьте остальные операторы сравнения